### DISCUSSION TITLE
Thoughts about sections inside the combobox listbox?

### DISCUSSION THREAD

**Discussion by heisand** [adobe/react-spectrum#4910] [Q&A] [answered]
I have a question about the sections within a combobox listbox. Today they are aria-hidden, but the corresponding group is aria-labelledby the section. 

When I navigate with a screen reader into the listbox, it is only announced that I am in a group inside a listbox. Only if I navigate to the end of the group, or navigate back from the first element, it is announced that I am currently in the "section name" group. In other words, one cannot really tell what section you are in before navigating all the way to the end, or back. Have you made any thoughts about the best way to announce the section you are currently navigating in?

**Comment by LFDanLu** [comment]
So our thought behind this behavior was that it would be too noisy if the current section was announced every time you moved focus from item to item. As a compromise, we decided to only announce the current section if the user was entering it from another section since then the user would theoretically have enough context for subsequent item focus changes. Note that this is specifically if the user was navigating via keyboard arrow keys to move focus up/down through the list. If the user were to move the accessibility cursor via an alternative method (aka mobile screen reader left/right swipes, control + option + arrow keys in VO, etc) the idea was that they would have to swipe past the section header which would then get announced.

Did the section/group not get announced the first time you moved focus into the listbox? Also the section headers themselves aren't aria-hidden I believe, mind sharing where you are seeing this?

**Reply by heisand** [reply]
I will try to add some screenshots to illustrate what I see.

The first two is what being announced first. 1. "Inside listbox", 2. "You are on a group, inside a listbox". 
The third is what being announced not until I start navigating backwards (or if I had navigated forwards to the end of the group).

I test the example from https://codesandbox.io/s/hardcore-moon-xzc4r?file=/src/ListBox.tsx, with VoiceOver on MacOS in Chrome. I navigate using control + option + arrow keys, and by doing so the only elements one can swipe past are the selectable options (in my case "Yambee" and "Abbie Binyon"). In this case I therefore do not know what section/group I am currently in upon the first time moving into the listbox.

The last code snippet shows the aria-hidden section.

 I hope this provides some more context here.

1. 
![Skjermbilde 2023-08-17 kl  11 30 55](https://github.com/adobe/react-spectrum/assets/15145686/afda1d33-6050-4c95-8d56-fec448d47dd2)

2. 
![Skjermbilde 2023-08-17 kl  10 49 48](https://github.com/adobe/react-spectrum/assets/15145686/5720577c-7039-4725-8a10-bbbd2343d4c6)

3. 
![Skjermbilde 2023-08-17 kl  09 23 56](https://github.com/adobe/react-spectrum/assets/15145686/c999d531-7770-45e0-b7b4-24b3a7f01a89)



        <li role="presentation" class="pt-2"><span id="react-aria5335816461-730" aria-hidden="true"
                class="text-xs font-bold uppercase text-gray-500 mx-3">Companies</span>
            <ul role="group" aria-labelledby="react-aria5335816461-730">

**Reply by snowystinger** [reply]
Just a note, while aria-hidden is true on the span, the group role is labelledby it, so it can still be read. It just can't be visited by the virtual cursor.

**Reply by heisand** [reply]
I know that the group is labelledby it, the problem is that the section is not announced right away (as explained in the description).

**Reply by LFDanLu** [reply]
Also its strange that the header there has `aria-hidden="true"`, none of our other examples of ListBoxes/ComboBox feature that (some have slightly different implementations and don't use `ul` and `li`): 
https://react-spectrum.adobe.com/react-aria/useListBox.html#sections
https://react-spectrum.adobe.com/react-spectrum/ComboBox.html#sections
https://react-spectrum.adobe.com/react-aria/ComboBox.html#sections
https://react-spectrum.adobe.com/react-aria/ListBox.html#sections

**Reply by LFDanLu** [reply, marked as answer]
@heisand Ah, looks like that sandbox is using an out of date version of `useListBoxSection`, bumped the package versions to latest: https://codesandbox.io/s/blazing-fog-shhzwh?file=/src/SearchAutocomplete.tsx:2402-2408

This was fixed via https://github.com/adobe/react-spectrum/pull/4037, we should bump the package versions in those sandboxes, thanks for bringing this to our attention. Does that sandbox exhibit the correct behavior for you (aka the section headers are now focusable via the screen reader cursor?)

**Reply by heisand** [reply]
Oh, great, let me do a quick test here.

**Reply by heisand** [reply]
It looks better now, thanks. I might have other related questions later, but will ask them separately 👍
